### ISSUE TITLE
How many Rays per iteration to pick?

### ISSUE THREAD

**Issue by malfonsoarquimea** [facebookresearch/Neural_3D_Video#10]
As far as I understood, for each iteration you select all the rays (or pixels) of all the views of a given frame and then you select some of those rays according to the weights generated (ISG or IST depending on the training stage). What I have been unable to find in the paper is how many rays do you pick per iteration. According to some calculations with the training times in the paper, it should be something like 1/5500 of all the available rays. Is that correct?
Also, do you pick those rays with or without replacement?
Thanks in advance!

**Comment by zhaoyang-lv** [comment]
The number of rays in a batch depending on how many rays you can feed into gpus and how many gpus you use. We did not find it matter quite much as long as the training loss can go down smoothly. We used 1024 rays per gpu in a majority of default experiments. The training is mostly stable for 4 gpus which result in 1024*4 rays per iteration.

**Comment by malfonsoarquimea** [comment]
> The number of rays in a batch depending on how many rays you can feed into gpus and how many gpus you use. We did not find it matter quite much as long as the training loss can go down smoothly. We used 1024 rays per gpu in a majority of default experiments. The training is mostly stable for 4 gpus which result in 1024*4 rays per iteration.

thanks very much for your answer!Do you pick rays with or without replacement? (can a ray appear twice in an iteration?)
Also, did you notice that fewer iterations are needed to reach the desired quality when the size of the trained images is smaller?
thanks for your time  @zhaoyang-lv
